server/aesdsocket.c

```diff
@@ -37,36 +124,268 @@ static void signal_handler (int signo)
         if (signo == SIGINT || signo == SIGTERM)
 	{
 		syslog(LOG_DEBUG, "Caught Signal, exiting");
-		close(socket_fd);
-		close(client_fd);
-	        close(write_fd);
-		closelog();
-		free(buffer);
-		free(transmit_buffer);
-		remove(FILE_PATH);
+		shutdown(socket_fd,SHUT_RDWR);
+		stop = 1;
+		close_all();
+		exit (EXIT_SUCCESS);
```

Comment:
@midhun9 this makes the `stop` flag redundant.  It would be better to use the `stop` flag.